🪟🐛 Connector form: Remove empty strings from form values for optional properties

flash1293 opened the pull request:
## What

Fixes https://github.com/airbytehq/airbyte/issues/8060

## How

If a user is entering a value in an optional field and removes it subsequently, an empty string will reside in that property. This PR detects this situation and overrides the value with undefined in the same react commit.

This is done for regular inputs, textareas, secret inputs and textareas as well as date(-time) inputs. It does not happen for array-inputs and enums.

josephkmh commented on airbyte-webapp/src/views/Connector/ConnectorForm/components/Property/Control.tsx:
```diff
@@ -25,6 +25,18 @@ export const Control: React.FC<ControlProps> = ({ property, name, disabled, erro
+  const onChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      field.onChange(e);
+      if (!property.isRequired && e.target.value === "") {
```
What's the reason for only doing this for optional fields? Could we not also set required fields to `undefined` if a user deletes the value?

flash1293 replied:
Given that an empty string will fail the yup validation anyway for required properties you are right that there is no difference in practice. Simplified the condition.

josephkmh reviewed (approved):
LGTM 👍

flash1293 commented:
@josephkmh glad you made the comment with the required field - the failing tests are an actual failure because if a field is set to undefined, formik will automatically fill in the default value again if there is one (this broke entering the port number in the e2e test).

I fixed this case by not setting a field to undefined if there is a default value as formik will fill it in when casting so it won't send an empty string in this case either. Added it to the list of things that are annoying about formik...

josephkmh commented:
Good catch! That is a strange behavior, TIL. Never underestimate formik 😭
